### DISCUSSION TITLE
AlbumArtist/Artist search issue - how to work around?

### DISCUSSION THREAD

**Discussion by GK-774** [jcorporation/myMPD#806] [Q&A] [answered]
Sorry for bringing this up again although it was already discussed in https://github.com/jcorporation/myMPD/issues/791 and https://github.com/jcorporation/myMPD/issues/804 which were closed for some reason.

Just tried it for a band I have 12 albums from and the search shows only the three that have AlbumArtist set, the others are silently ignored, which is unexpected and a deal breaker for larger collections.
To get the search working properly, where would I need to disable AlbumArtist, in "Tags to use", "Tags to search", "Tags to browse" or all of them?

In my view "AlbumArtist" is just an additional tag (nice for compilations) but not a replacement of "Artist". Expected behaviour, in my view, would be, if I search "any tag" then _any_ tag shall be actually searched. Current state is, "any" does not mean "any" but "any tag except for Artist if you have enabled AlbumArtist".

**Comment by jcorporation** [comment, marked as answer]
My last comment of #791 should answer your question.

Workaround for 9.4.x is to disable the albumartist tag in „tags to use“

**Comment by GK-774** [comment]
Will the fix mentioned in https://github.com/jcorporation/myMPD/issues/791 be implemented in 9.4.2 as well? Or is 9.3.4 basically the stable release for now?

**Reply by jcorporation** [reply]
Unsure, the fix can not be easily backported. If you can not use the workaround you should stick on 9.3.4 until 9.5.0 is released or is there any requirement to go to 9.4.1?

**Comment by GK-774** [comment]
No, no special requirement. 9.3.4 works perfectly fine (and did so since it was released). With 9.4.1 and AlbumArtist off, I can see all albums, but somehow when sorting by "Last modified" (for the newest albums) there is a mixture of recent and old albums. I have not investigated this in detail due to current lack of time but got the same effect with 9.5.0 and AlbumArtist on.

**Comment by jcorporation** [comment]
Determining the album last modified timestamp was improved.

Old: use the last modification of first song that was found for an album, this can be any song in the album, mostly track 1.

New: use the newest modification date from all songs in the album.

**Comment by jcorporation** [comment]
myMPD v9.5.0 is released. I hope it fixes all issues with the album view.

**Comment by GK-774** [comment]
Just tried 9.5.1 and whatever timestamp it now uses, it does definitely not show the albums which were recently added when using "Last modified" "Descending". With 9.3.4 I get the last album added is shown first and so on in descending order, while 9.5.1 shows me albums where I have no idea when they where added, certainly not recently.

**Comment by jcorporation** [comment]
I checked it with my music collection and it works for me. Could you cross check the last modified dates? You can add a last modified column in the album detail view.

**Comment by GK-774** [comment]
I now tried this with 9.3.4 compared to 9.5.1

"newest" is Album View, Last modified, Descending
"oldest" is Album View, Last modified, Ascending ("Descending" unticked)

I cross checked the times on the file system (ls -l) and the date and time values match.
I did a "clear app cache and reload" after the installaton.

9.3.4
newest: 8/10/2022 11:52:21 AM
oldest: 4/16/2001 1:04:00 PM
Looks plausible, and the newest albums are, as far I remember, in the order I have added them.

9.5.1
newest: 6/29/2001 5:56:48 PM
oldest: 9/29/2001 9:27:38 AM
Looks like reading the time stamps (as they match the file system) works ok, but sorting does not.

Hope this helps, I cannot contribute further tests until Aug 30.

**Reply by jcorporation** [reply]
Sorting algorithm has indeed changed in v9.4.0. Now myMPD uses a radix tree for sorting, this is much faster than the old sorting of a linked list.

Timestamps are not read from the filesystem. Last-modified is the timestamp mpd has added/changed the song in his database.

**Comment by GK-774** [comment]
Well, ok, but when I add an album it gets on the file system and then I do an update (so the mpd database time stamp is a little bit newer). At any rate, "newest: 6/29/2001" certainly is wrong in any case while "newest: 8/10/2022" is correct.

**Reply by jcorporation** [reply]
Commit 7b071c230adca1398827966c48011b8658291966 adds a fix that makes the sort key equal width, adding 0-padding in front.

```
date +%s --date "6/29/2001"
993765600

date +%s --date "8/10/2022"
1660082400
```

This should fix your issue.

**Reply by GK-774** [reply]
Yes, it works as expected now. Many thanks!
